sdks/python/apache_beam/runners/interactive/interactive_runner.py

```diff
@@ -117,123 +119,43 @@ def apply(self, transform, pvalueish, options):
     return self._underlying_runner.apply(transform, pvalueish, options)
 
   def run_pipeline(self, pipeline, options):
-    if not hasattr(self, '_desired_cache_labels'):
-      self._desired_cache_labels = set()
-
-    # Invoke a round trip through the runner API. This makes sure the Pipeline
-    # proto is stable.
-    pipeline = beam.pipeline.Pipeline.from_runner_api(
-        pipeline.to_runner_api(use_fake_coders=True),
-        pipeline.runner,
-        options)
-
-    # Snapshot the pipeline in a portable proto before mutating it.
-    pipeline_proto, original_context = pipeline.to_runner_api(
-        return_context=True, use_fake_coders=True)
-    pcolls_to_pcoll_id = self._pcolls_to_pcoll_id(pipeline, original_context)
-
-    analyzer = pipeline_analyzer.PipelineAnalyzer(self._cache_manager,
-                                                  pipeline_proto,
-                                                  self._underlying_runner,
-                                                  options,
-                                                  self._desired_cache_labels)
-    # Should be only accessed for debugging purpose.
-    self._analyzer = analyzer
+    pin = inst.pin(pipeline, options)
 
     pipeline_to_execute = beam.pipeline.Pipeline.from_runner_api(
-        analyzer.pipeline_proto_to_execute(),
+        pin.instrumented_pipeline_proto(),
         self._underlying_runner,
         options)
 
     if not self._skip_display:
-      display = display_manager.DisplayManager(
-          pipeline_proto=pipeline_proto,
-          pipeline_analyzer=analyzer,
-          cache_manager=self._cache_manager,
-          pipeline_graph_renderer=self._renderer)
-      display.start_periodic_update()
+      pg = pipeline_graph.PipelineGraph(pin.original_pipeline,
```

Comment:
Let's rename this into something more descriptive.

Reply:
Sure! I'll use the full name `pipeline_graph`.